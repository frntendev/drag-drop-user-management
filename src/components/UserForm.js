import React from "react";
import { Field } from "redux-form";
import { css } from "react-emotion";
import TextField from "material-ui/TextField";
const renderTextField = ({
  input,
  label,
  meta: { touched, error },
  ...custom
}) => (
  <TextField
    label={label}
    error={touched && error ? true : false}
    {...input}
    {...custom}
  />
);
const required = value => (value ? undefined : "Required");
const UserForm = props => {
  return (
    <div
    id="user-form"
      className={css`
        display: flex;
        flex-wrap: wrap;
      `}
    >
      <Field
        name="firstname"
        component={renderTextField}
        label="First Name"
        validate={[required]}
        className={css`
          flex: 1;
          margin-right: 5%;
        `}
      />
      <Field
        name="lastname"
        component={renderTextField}
        label="Last Name"
        validate={[required]}
        className={css`
          flex: 1;
        `}
      />
      <Field
        name="allegiances"
        component={renderTextField}
        label="Allegiances"
        validate={[required]}
        fullWidth
        className={css`
          margin-top: 2%;
        `}
      />
    </div>
  );
};

export default UserForm;
