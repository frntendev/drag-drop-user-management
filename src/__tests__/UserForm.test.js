import React from "react";
import { shallow, mount, configure } from "enzyme";
import { Field } from "redux-form";
import UserForm from "../components/UserForm";
import Adapter from "enzyme-adapter-react-15";
import TextField from "material-ui/TextField";
import renderer from "react-test-renderer";

configure({ adapter: new Adapter() });
const renderTextField = ({
  input,
  label,
  meta: { touched, error },
  ...custom
}) => (
  <TextField
    label={label}
    error={touched && error ? true : false}
    {...input}
    {...custom}
  />
);
describe("UserForm Component", () => {
  const shallowComponent = shallow(<UserForm />);
  // const mountComponent = mount(<GroupForm />);
  it("should render without throwing an error", () => {
    expect(shallowComponent).toMatchSnapshot();
  });
  it("should find textfields in the component", () => {
    expect(shallowComponent.find("#user-form").length).toEqual(1);
    expect(shallowComponent.find("[name='firstname']").length).toEqual(1);
    expect(shallowComponent.find("[name='lastname']").length).toEqual(1);
    expect(shallowComponent.find("[name='allegiances']").length).toEqual(1);
  });
});
