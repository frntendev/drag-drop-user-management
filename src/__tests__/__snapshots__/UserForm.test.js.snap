// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm Component should render without throwing an error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Field
          className="css-1ose7zb"
          component={[Function]}
          label="First Name"
          name="firstname"
          validate={
            Array [
              [Function],
            ]
          }
        />,
        <Field
          className="css-r6z50o"
          component={[Function]}
          label="Last Name"
          name="lastname"
          validate={
            Array [
              [Function],
            ]
          }
        />,
        <Field
          className="css-10nkwyu"
          component={[Function]}
          fullWidth={true}
          label="Allegiances"
          name="allegiances"
          validate={
            Array [
              [Function],
            ]
          }
        />,
      ],
      "className": "css-ak0z00",
      "id": "user-form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "css-1ose7zb",
          "component": [Function],
          "label": "First Name",
          "name": "firstname",
          "validate": Array [
            [Function],
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "css-r6z50o",
          "component": [Function],
          "label": "Last Name",
          "name": "lastname",
          "validate": Array [
            [Function],
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "css-10nkwyu",
          "component": [Function],
          "fullWidth": true,
          "label": "Allegiances",
          "name": "allegiances",
          "validate": Array [
            [Function],
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Field
            className="css-1ose7zb"
            component={[Function]}
            label="First Name"
            name="firstname"
            validate={
              Array [
                [Function],
              ]
            }
          />,
          <Field
            className="css-r6z50o"
            component={[Function]}
            label="Last Name"
            name="lastname"
            validate={
              Array [
                [Function],
              ]
            }
          />,
          <Field
            className="css-10nkwyu"
            component={[Function]}
            fullWidth={true}
            label="Allegiances"
            name="allegiances"
            validate={
              Array [
                [Function],
              ]
            }
          />,
        ],
        "className": "css-ak0z00",
        "id": "user-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "css-1ose7zb",
            "component": [Function],
            "label": "First Name",
            "name": "firstname",
            "validate": Array [
              [Function],
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "css-r6z50o",
            "component": [Function],
            "label": "Last Name",
            "name": "lastname",
            "validate": Array [
              [Function],
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "css-10nkwyu",
            "component": [Function],
            "fullWidth": true,
            "label": "Allegiances",
            "name": "allegiances",
            "validate": Array [
              [Function],
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
